### PR TITLE
Feat: Delete Downloaded Deck after successful import

### LINKED ISSUE #10663: Delete deck after download and import successfully
**Is your feature request related to a problem? Please describe.**
The deck remains in `Download` folder after download and import using `Get Shared Deck`. So it needs to clear the downloaded deck file to free up some space.

**Describe the solution you'd like**
Clear the successfully downloaded and imported deck from Download folder

**Describe alternatives you've considered**
Although the folder `Android/data/com.ichi2.anki/files/Download` is not accessible in normal File Explorer, so used `Files` to access and delete file.

**Additional context**

### PR DESCRIPTION
**Pull request by Kshitij09**
## Purpose / Description
As briefly discussed in #10663, Shared Decks are downloaded to the app-specific Download directory (`Android/data/com.ichi2.anki/files/Download`). After successful import, these files remain in there, adding to App's Storage and not required by any Use Case as well. This PR adds cleanup implementation at callback receiving 'Import Successful'.

## Fixes
Fixes #10663

## Approach
After downloading the deck file, the same is copied to cache directory and all the further import operations are implemented using this version. As a result, on the UI site, we only get access to cache file which we're not supposed to clean up. This PR adds an implementation that derives Downloaded Uri from the cache FileName and makes the required `delete(Uri..)` call on the `ContentResolver`
## How Has This Been Tested?
Tested on Pixel 4a running Android 12 (API 31)

## Learning (optional, can help others)
How to delete files using ContentResolver ([stackoverflow](https://stackoverflow.com/a/43563972))

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [X] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [X] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by Kshitij09** [comment]
@krmanik @ShridharGoel Kindly do review the changes. There's a scope for refactoring and am open to your suggestions.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 1]
```diff
@@ -345,10 +351,10 @@ public void actualOnPostExecute(@NonNull DeckPicker deckPicker, Triple<AnkiPacka
                 AnkiPackageImporter imp = result.first;
                 deckPicker.showSimpleMessageDialog(TextUtils.join("\n", imp.getLog()));
                 deckPicker.updateDeckList();
+                deckPicker.cleanupDownloadedDeck(mImportPath);
```
Comment (krmanik) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:354]: It is better to call this function when the `import` called from `shared deck` fragments. May be intent will be used to know, from where the activity/fragments opened.

Comment (krmanik) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:354]: When we use chrome to download decks from other source like drive, then decks remain in download folder `Android/0/emulated/Download`.  But in AnkiDroid, when deck downloaded using `Get Shared Decks` then it downloaded in  `Android/data/com.ichi2.anki/files/Download`.  So here the file should be removed from `Android/data/com.ichi2.anki/files/Download` folder. 

The import dialog is opened when `.apkg` file open in file manager, so code in this PR will remove the deck file from `Android/0/emulated/Download` also. It should not remove this file.

Comment (Kshitij09) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:354]: So basically we are only interested in deleting the decks that we downloaded from "Shared Decks", IE, source triggering the import dialog should be `SharedDecksFragment` and all other sources should be kept intact

**Pull request closed without merging**